server/client.js

```diff
@@ -70,22 +70,33 @@ async function sendHeartbeatList(socket, monitorID, toUser = false, overwrite =
 /**
  * Important Heart beat list (aka event list)
  * @param {Socket} socket Socket.io instance
- * @param {number} monitorID ID of monitor to send heartbeat history
+ * @param {number} monitorID ID of monitor to send heartbeat history, or null for all monitors
  * @param {boolean} [toUser=false]  True = send to all browsers with the same user id, False = send to the current browser only
  * @param {boolean} [overwrite=false] Overwrite client-side's heartbeat list
  * @returns {Promise<void>}
  */
 async function sendImportantHeartbeatList(socket, monitorID, toUser = false, overwrite = false) {
     const timeLogger = new TimeLogger();
 
-    let list = await R.find("heartbeat", `
+    let list = [];
+
+    if (monitorID == null) {
+        // Send important beats for all monitors
+        list = await R.find("heartbeat", `
+        important = 1
```

Comment:
I don't think these are functionally identical:
This is finding the top 5k important heartbeats while the other one is finding the top 500 important heartbeats **for each monitor**.

=> Lets assume a monitor runs every 20s and every heartbeat is important => 4320 beats
=> have 2 monitors which produce beats like this and one which pings once per day and the results will not be identical
=> I think this is missing a group by

Reply:
I see your point that if the important beat list grew too long, it would lead to different behavior. I'd say for the initial dashboard load, 5000 is more than enough. That's 250 pages and no one will bother going through the unsearchable list. I think I have two options here:

- Add a function to send the important heartbeat list when the user clicks on an entry in the dashboard, in case there are any missing, or
- Modify the SQL to replicate the original behavior. A quick search and the only way to do it seems to be a nested SELECT, looks a bit ugly.

Why would GROUP BY be needed here when we are not using an aggregate function?